pkg/kubelet/kubelet_getters.go

```diff
@@ -228,12 +228,10 @@ func (kl *Kubelet) GetNode() (*v1.Node, error) {
 // in which case return a manufactured nodeInfo representing a node with no pods,
 // zero capacity, and the default labels.
 func (kl *Kubelet) getNodeAnyWay() (*v1.Node, error) {
-	if kl.kubeClient != nil {
-		if n, err := kl.nodeInfo.GetNodeInfo(string(kl.nodeName)); err == nil {
-			return n, nil
-		}
+	if kl.kubeClient == nil {
+		return kl.initialNode(false)
 	}
-	return kl.initialNode()
+	return kl.nodeInfo.GetNodeInfo(string(kl.nodeName))
```

Comment:
This is changing the logic, since it won't fall back to initialNode in case of an error.

Reply:
Oops. Nice catch.